pkg/co2/carbon_intensity.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package co2
+
+import (
+	"fmt"
+
+	co2_source "github.com/sustainable-computing-io/kepler/pkg/co2/source"
+	"github.com/sustainable-computing-io/kepler/pkg/config"
+)
+
+var (
+	co2Impl co2Interface = &co2_source.UKGrid{}
+	errLib               = fmt.Errorf("could not start carbon intensity collector")
+)
+
+type co2Interface interface {
+	// Init initizalize and start the CO2 metric collector
+	Init() error
+	// Shutdown stops the CO2 metric collector
+	Shutdown() bool
+	// GetCarbonIntensity returns a value of mg/joule CO2
+	GetCarbonIntensity(string) (int64, error)
+	// IsCO2CollectionSupported returns if it is possible to use this collector
+	IsCO2CollectionSupported() bool
+	// SetCO2CollectionSupported manually set if it is possible to use this collector. This is for testing purpose only.
+	SetCO2CollectionSupported(bool)
+}
+
+func Init() error {
+	return errLib
+}
+
+func Shutdown() bool {
+	if co2Impl != nil && config.EnabledCO2 {
+		return co2Impl.Shutdown()
+	}
+	return true
+}
+
+func GetCarbonIntensity(url string) (int64, error) {
+	if co2Impl != nil && config.EnabledCO2 {
+		return co2Impl.GetCarbonIntensity("https://api.carbonintensity.org.uk/intensity")
```

Comment:
May I have a brief for this API?

Reply:
What sort of format are you looking for? I think this PR is more a discussion to design a brief.

Reply:
@jbu , when I access this API, it responses as below:
```
{ 
  "data":[{ 
    "from": "2023-05-14T11:00Z",
    "to": "2023-05-14T11:30Z",
    "intensity": {
      "forecast": 121,
      "actual": 112,
      "index": "low"
    }
  }]
}
```
Hence serval questions:
1. what does this API mean?
2. what does this json format means?
3. according to `org.uk` ? so it's some data from which org? and considering other regions of world, any other endpoints?
etc....
you know, for any one without any background knowledge about this API to review this part of code, there are lots of question marks in their mind. cc: @rootfs